src/Avalonia.Controls/SplitButton/SplitButton.cs

```diff
@@ -0,0 +1,709 @@
+        ":checked",
+        ":checkedPrimaryPressed",
+        ":checkedPrimaryPointerOver",
+        ":checkedSecondaryPressed",
+        ":checkedSecondaryPointerOver",
+        ":touchPressed",
+        ":primaryPressed",
+        ":primaryPointerOver",
+        ":secondaryPressed",
+        ":secondaryPointerOver")]
+    public class SplitButton : ContentControl, ICommandSource
+    {
+        /// <summary>
+        /// Raised when the user presses the primary part of the <see cref="SplitButton"/>.
+        /// </summary>
+        public event EventHandler<RoutedEventArgs> Click
+        {
+            add => AddHandler(ClickEvent, value);
+            remove => RemoveHandler(ClickEvent, value);
+        }
+
+        /// <summary>
+        /// Defines the <see cref="Click"/> event.
+        /// </summary>
+        public static readonly RoutedEvent<RoutedEventArgs> ClickEvent =
+            RoutedEvent.Register<SplitButton, RoutedEventArgs>(
+                nameof(Click),
+                RoutingStrategies.Bubble);
+
+        /// <summary>
+        /// Defines the <see cref="Command"/> property.
+        /// </summary>
+        public static readonly DirectProperty<SplitButton, ICommand> CommandProperty =
+            AvaloniaProperty.RegisterDirect<SplitButton, ICommand>(
+                nameof(Command),
+                splitButton => splitButton.Command,
+                (splitButton, command) => splitButton.Command = command,
+                enableDataValidation: true);
+
+        /// <summary>
+        /// Defines the <see cref="CommandParameter"/> property.
+        /// </summary>
+        public static readonly StyledProperty<object> CommandParameterProperty =
+            AvaloniaProperty.Register<SplitButton, object>(
+                nameof(CommandParameter));
+
+        /// <summary>
+        /// Defines the <see cref="Flyout"/> property
+        /// </summary>
+        public static readonly StyledProperty<FlyoutBase> FlyoutProperty =
+            AvaloniaProperty.Register<SplitButton, FlyoutBase>(
+                nameof(Flyout));
+
+        private ICommand _Command;
+
+        private Button _primaryButton   = null;
+        private Button _secondaryButton = null;
+
+        private bool        _commandCanExecute       = true;
+        protected bool      _hasLoaded               = false;
+        private bool        _isAttachedToLogicalTree = false;
+        private bool        _isFlyoutOpen            = false;
+        private bool        _isKeyDown               = false;
+        private PointerType _lastPointerType         = PointerType.Mouse;
+
+        private CompositeDisposable _buttonPropertyChangedDisposable;
+        private IDisposable         _flyoutPropertyChangedDisposable;
+
+        ////////////////////////////////////////////////////////////////////////
+        // 
+        // Constructor / Destructors
+        //
+        ////////////////////////////////////////////////////////////////////////
```

Comment:
Not sure if we need these huge blocks. Do we have them anywhere?

Reply:
Nope, section headings based on an internal standard. Figured someone might complain :) The only concern I have is the code is neatly organized now in clear sections. Without the headings no one will think to notice that in the future.

Reply:
I reduced the size of these section headings. Might look better to you now.